src/Common/JemallocNodumpAllocatorImpl.cpp

```diff
@@ -0,0 +1,84 @@
+#include <Common/JemallocNodumpAllocatorImpl.h>
+
+#if USE_JEMALLOC
+#include <Poco/Format.h>
+#include <base/errnoToString.h>
+#include <base/defines.h>
+#include <sys/mman.h>
+#include <cstring>
+
+namespace DB
+{
+
+JemallocNodumpAllocatorImpl & JemallocNodumpAllocatorImpl::instance()
+{
+    static JemallocNodumpAllocatorImpl allocator;
+    return allocator;
+}
+
+void * JemallocNodumpAllocatorImpl::allocate(size_t size, size_t alignment) const
+{
+    return mallocx(size, alignment ? flags | MALLOCX_ALIGN(alignment) : flags);
```

Comment:
Pardon my ignorance about Jemalloc. How's this mallocx glued to end up using the same arena that is allocated in `alloc`?

I expected to see here `MALLOCX_ARENA(arena_index)` or something similar

Reply:
I missed that `flags` was a class member, so I was puzzled it worked :sweat: